s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -0,0 +1,141 @@
+                .withWebIdentityToken(credentials.getOauth().getAccessToken());
+
+        String durationsecondsProp = "s3.assumerole.durationseconds";
+        String policyProp = "s3.assumerole.policy";
+        String rolearnProp = "s3.assumerole.rolearn";
+        String rolesessionnameProp = "s3.assumerole.rolesessionname";
+
+        if (StringUtils.isBlank(host.getProperty(durationsecondsProp))) {
+            webIdReq = webIdReq.withDurationSeconds(Integer.getInteger(host.getProperty(durationsecondsProp)));
+        }
+        else if(host.getProtocol().getProperties().containsKey(durationsecondsProp)) {
+            System.out.println("set duration second with profile value: " + host.getProtocol().getProperties().get(durationsecondsProp));
+            webIdReq = webIdReq.withDurationSeconds(Integer.getInteger(host.getProtocol().getProperties().get(durationsecondsProp)));
+        }
+        else if(StringUtils.isNotBlank(PreferencesFactory.get().getProperty(durationsecondsProp))) {
+            webIdReq = webIdReq.withDurationSeconds(new HostPreferences(host).getInteger(durationsecondsProp));
+        }
+
+        if (StringUtils.isBlank(host.getProperty(policyProp))) {
+            webIdReq = webIdReq.withPolicy(host.getProperty(policyProp));
+        }
+        else if(host.getProtocol().getProperties().containsKey(policyProp)) {
+            webIdReq = webIdReq.withPolicy(host.getProtocol().getProperties().get(policyProp));
+        }
+        else if(StringUtils.isNotBlank(PreferencesFactory.get().getProperty(policyProp))) {
+            webIdReq = webIdReq.withPolicy(new HostPreferences(host).getProperty(policyProp));
+        }
+
+        if (StringUtils.isBlank(host.getProperty(rolearnProp))) {
+            webIdReq = webIdReq.withRoleArn(host.getProperty(rolearnProp));
+        }
+        else if(host.getProtocol().getProperties().containsKey(rolearnProp)) {
+            webIdReq = webIdReq.withRoleArn(host.getProtocol().getProperties().get(rolearnProp));
+        }
+        else if(StringUtils.isNotBlank(PreferencesFactory.get().getProperty(rolearnProp))) {
+            webIdReq = webIdReq.withRoleArn(new HostPreferences(host).getProperty(rolearnProp));
+        }
+
+        if (StringUtils.isBlank(host.getProperty(rolesessionnameProp))) {
+            webIdReq = webIdReq.withRoleSessionName(host.getProperty(rolesessionnameProp));
+        }
+        else if(host.getProtocol().getProperties().containsKey(rolesessionnameProp)) {
+            webIdReq = webIdReq.withRoleSessionName(host.getProtocol().getProperties().get(rolesessionnameProp));
+        }
+        else if(StringUtils.isNotBlank(PreferencesFactory.get().getProperty(rolesessionnameProp))) {
+            webIdReq = webIdReq.withRoleSessionName(new HostPreferences(host).getProperty(rolesessionnameProp));
+        }
+
+        AssumeRoleWithWebIdentityResult result = service.assumeRoleWithWebIdentity(webIdReq);
```

Comment:
Missing exception handling for failures such as `InvalidIdentityTokenException`.

Reply:
There are a whole bunch of possible exceptions. What is the right reaction in the catch-statements?
`InvalidIdentityTokenException` -> oAuth-authorize
`ExpiredTokenException` -> oAuth-refresh
other Exception -> log error message?